src/utils/numericEncodingOfArray.ts

```diff
@@ -0,0 +1,27 @@
+/**
+ * Numerically encodes the strings in the array and returns an encoding dictionary which can be used to encode other arrays
+ * @param array - original array before encoding
+ * @returns - dictionnary from string to number
+ */
```

Comment:
The method that have as first parameter a matrix should be named starting with matrix and be moved in the folder matrix.